### PR TITLE
Use same assembly for single- and multiuser Che

### LINKED ISSUE #6981: [CHE6] Make single assembly for different infrastructures and modes
### Description
We would like to have single assembly that can be
- Running on docker on OpenShift
- Single user  vs Multi-user (PostgreSQL + Keycloak)

### PR DESCRIPTION
**Pull request by mshaposhnik**
### What does this PR do?
Removes separate assembly for multiuser version, removes some old stuff like machine token services/clients.

### What issues does this PR fix or reference?
fixes #6981

#### Release Notes
Use same  assembly for singleuser and multiuser Che.

#### Docs PR
N/A

### CODE REVIEW

File: assembly/assembly-wsmaster-war/src/main/java/org/eclipse/che/api/deploy/WsMasterServletModule.java [hunk 1]
```diff
@@ -46,5 +48,21 @@ protected void configureServlets() {
 
     serveRegex("^/(?!ws$|ws/|websocket.?)(.*)").with(GuiceEverrestServlet.class);
     install(new org.eclipse.che.swagger.deploy.BasicSwaggerConfigurationModule());
+
+    if (Boolean.valueOf(System.getenv("CHE_MULTIUSER"))) {
+      configureMultiUserMode();
+    } else {
```
Comment (riuvshin) [inline assembly/assembly-wsmaster-war/src/main/java/org/eclipse/che/api/deploy/WsMasterServletModule.java:54]: NORM

File: assembly/assembly-wsmaster-war/src/main/webapp/WEB-INF/classes/codenvy/multiuser.properties [hunk 2]
```diff
@@ -119,4 +119,4 @@ che.keycloak.oso.endpoint=NULL
 che.keycloak.github.endpoint=NULL
 
 #     The number of seconds to tolerate for clock skew when verifying exp or nbf claims.
-che.keycloak.allowed_clock_skew_sec=3
+che.keycloak.allowed_clock_skew_sec=3
```
Comment (riuvshin) [inline assembly/assembly-wsmaster-war/src/main/webapp/WEB-INF/classes/codenvy/multiuser.properties]: new line

File: assembly/assembly-wsmaster-war/src/main/java/org/eclipse/che/api/deploy/WsMasterModule.java [hunk 3]
```diff
@@ -193,4 +240,103 @@ protected void configure() {
...
+
+    bind(org.eclipse.che.api.user.server.CheUserCreator.class);
+
+    bindConstant().annotatedWith(Names.named("che.agents.auth_enabled")).to(false);
+    bindConstant()
+        .annotatedWith(Names.named("db.jndi.datasource.name"))
+        .to("java:/comp/env/jdbc/che-h2");
+
+    bindConstant()
+        .annotatedWith(Names.named("machine.terminal_agent.run_command"))
```
Comment (garagatyi) [inline assembly/assembly-wsmaster-war/src/main/java/org/eclipse/che/api/deploy/WsMasterModule.java]: Can you check whether it is used anywhere?

File: assembly/assembly-wsmaster-war/src/main/java/org/eclipse/che/api/deploy/WsMasterModule.java [hunk 4]
```diff
@@ -193,4 +240,103 @@ protected void configure() {
...
+        .to(
+            "$HOME/che/terminal/che-websocket-terminal "
+                + "-addr :4411 "
+                + "-cmd ${SHELL_INTERPRETER} "
+                + "-enable-activity-tracking");
+    bindConstant()
+        .annotatedWith(Names.named("machine.exec_agent.run_command"))
```
Comment (garagatyi) [inline assembly/assembly-wsmaster-war/src/main/java/org/eclipse/che/api/deploy/WsMasterModule.java]: Can you check whether it is used anywhere?

File: assembly/assembly-wsmaster-war/src/main/java/org/eclipse/che/api/deploy/WsMasterModule.java [hunk 5]
```diff
@@ -193,4 +240,103 @@ protected void configure() {
...
+    bindConstant().annotatedWith(Names.named("che.agents.auth_enabled")).to(true);
+
+    bindConstant()
+        .annotatedWith(Names.named("db.jndi.datasource.name"))
+        .to("java:/comp/env/jdbc/che-pg");
+
+    bindConstant()
+        .annotatedWith(Names.named("machine.terminal_agent.run_command"))
+        .to(
+            "$HOME/che/terminal/che-websocket-terminal "
```
Comment (garagatyi) [inline assembly/assembly-wsmaster-war/src/main/java/org/eclipse/che/api/deploy/WsMasterModule.java]: Can you check whether it is used anywhere?

File: assembly/assembly-wsmaster-war/src/main/java/org/eclipse/che/api/deploy/WsMasterModule.java [hunk 6]
```diff
@@ -193,4 +240,103 @@ protected void configure() {
...
+                + "-addr :4411 "
+                + "-cmd ${SHELL_INTERPRETER} "
+                + "-enable-auth "
+                + "-enable-activity-tracking");
+    bindConstant()
+        .annotatedWith(Names.named("machine.exec_agent.run_command"))
+        .to(
+            "$HOME/che/exec-agent/che-exec-agent "
```
Comment (garagatyi) [inline assembly/assembly-wsmaster-war/src/main/java/org/eclipse/che/api/deploy/WsMasterModule.java]: Can you check whether it is used anywhere?

File: multiuser/machine-auth/che-multiuser-machine-authentication/src/main/java/org/eclipse/che/multiuser/machine/authentication/server/MachineTokenPermissionsFilter.java [hunk 7]
```diff
@@ -1,59 +0,0 @@
...
-import org.everrest.core.resource.GenericResourceMethod;
-
-/**
- * Restricts access to methods of {@link MachineTokenService} by user's permissions
- *
- * @author Max Shaposhnik (mshaposhnik@codenvy.com)
- */
-@Filter
-@Path("/machine/token{path:(/.*)?}")
-public class MachineTokenPermissionsFilter extends CheMethodInvokerFilter {
```
Comment (garagatyi) [inline multiuser/machine-auth/che-multiuser-machine-authentication/src/main/java/org/eclipse/che/multiuser/machine/authentication/server/MachineTokenPermissionsFilter.java:32]: Looks like `MachineTokenService` still exists, but not protected. Can we remove it or protect?

**Review by garagatyi — APPROVED** [review]

File: multiuser/machine-auth/che-multiuser-machine-authentication/src/main/java/org/eclipse/che/multiuser/machine/authentication/server/MachineTokenPermissionsFilter.java (hunk 7 above)
Comment (mshaposhnik) [reply multiuser/machine-auth/che-multiuser-machine-authentication/src/main/java/org/eclipse/che/multiuser/machine/authentication/server/MachineTokenPermissionsFilter.java:32]: Forgot to remove, fixed

**Review by riuvshin — APPROVED** [review]
LGTM. but please before merge let me know because these changes will impact CI and I will need to change CI configuration.

**Review by skabashnyuk — APPROVED** [review]

File: multiuser/machine-auth/che-multiuser-machine-authentication-agent/src/main/java/org/eclipse/che/multiuser/machine/authentication/agent/MachineLoginFilter.java [hunk 8]
```diff
@@ -84,7 +84,7 @@ public void doFilter(ServletRequest request, ServletResponse response, FilterCha
     try {
       final UserDto userDescriptor =
           requestFactory
-              .fromUrl(tokenServiceEndpoint + "/user/" + machineToken)
+              .fromUrl(apiEndpoint + "/user/")
```
Comment (sleshchenko) [inline multiuser/machine-auth/che-multiuser-machine-authentication-agent/src/main/java/org/eclipse/che/multiuser/machine/authentication/agent/MachineLoginFilter.java:87]: I think now you should change catching of `NotFoundException` to catching of `UnauthorizedException`

File: multiuser/machine-auth/che-multiuser-machine-authentication-ide/src/main/java/org/eclipse/che/multiuser/machine/authentication/ide/MachineAsyncRequestFactory.java [hunk 9]
```diff
@@ -115,10 +76,14 @@ protected boolean isWsAgentRequest(String url) {
     if (currentWorkspace == null || !isWsAgentStarted(currentWorkspace)) {
       return false; // ws-agent not started
     }
-    return url.contains(nullToEmpty(appContext.getWsAgentServerApiEndpoint()));
+    final boolean contains = url.contains(nullToEmpty(appContext.getWsAgentServerApiEndpoint()));
```
Comment (sleshchenko) [inline multiuser/machine-auth/che-multiuser-machine-authentication-ide/src/main/java/org/eclipse/che/multiuser/machine/authentication/ide/MachineAsyncRequestFactory.java]: Why do you need this variable?

File: multiuser/machine-auth/che-multiuser-machine-authentication-ide/src/main/java/org/eclipse/che/multiuser/machine/authentication/ide/inject/MachineAuthGinModule.java [hunk 10]
```diff
@@ -11,19 +11,12 @@
...
 
 import com.google.gwt.inject.client.AbstractGinModule;
-import com.google.inject.Singleton;
 import org.eclipse.che.ide.api.extension.ExtensionGinModule;
-import org.eclipse.che.multiuser.machine.authentication.ide.MachineTokenServiceClient;
-import org.eclipse.che.multiuser.machine.authentication.ide.MachineTokenServiceClientImpl;
 
 /** @author Anton Korneta */
 @ExtensionGinModule
 public class MachineAuthGinModule extends AbstractGinModule {
```
Comment (sleshchenko) [inline multiuser/machine-auth/che-multiuser-machine-authentication-ide/src/main/java/org/eclipse/che/multiuser/machine/authentication/ide/inject/MachineAuthGinModule.java]: Why do we need this module that binds nothing?

**Review by sleshchenko — APPROVED** [review]

**Comment by sleshchenko** [comment]
Is it still WIP?

**Pull request merged**